test/terraform_aws_s3_example_test.go

```diff
@@ -55,3 +55,52 @@ func TestTerraformAwsS3Example(t *testing.T) {
 	expectedStatus := "Enabled"
 	assert.Equal(t, expectedStatus, actualStatus)
 }
+
+// An example of how to test the Terraform module in examples/terraform-aws-s3-example using Terratest with an S3 bucket policy.
+func TestTerraformAwsS3ExampleWithPolicy(t *testing.T) {
+	t.Parallel()
+
+	// Give this S3 Bucket a unique ID for a name tag so we can distinguish it from any other Buckets provisioned
+	// in your AWS account
+	expectedName := fmt.Sprintf("terratest-aws-s3-example-%s", strings.ToLower(random.UniqueId()))
+
+	// Give this S3 Bucket an environment to operate as a part of for the purposes of resource tagging
+	expectedEnvironment := "Automated Testing"
+
+	// Pick a random AWS region to test in. This helps ensure your code works in all regions.
+	awsRegion := aws.GetRandomStableRegion(t, nil, nil)
+
+	terraformOptions := &terraform.Options{
+		// The path to where our Terraform code is located
+		TerraformDir: "../examples/terraform-aws-s3-example",
```

Comment:
Since this example is now shared across two tests, there will be conflicts in the terraform state (it uses local state, stored in the example folder). We could use `test_structure.CopyTerraformFolderToTemp` to resolve this (see https://github.com/gruntwork-io/terratest/blob/master/test/terraform_ssh_example_test.go#L22 for an example), but given the simplicity of the checks, I am inclined to keep our examples simple and merge the two tests. Can you consolidate the two tests and just run both bucket policy and versioning checks?

Reply:
Implemented